pkg/util/kubectl/kubectl.go

```diff
@@ -354,3 +358,75 @@ func UpdateDeployment(deploy *v1.Deployment) error {
 	_, err := client.AppsV1().Deployments(deploy.Namespace).Update(context.TODO(), deploy, metav1.UpdateOptions{})
 	return err
 }
+
+// PatchOwnerReferenceWithAppInfoFile patch tfjob / pytorchjob ownerReference
+func PatchOwnerReferenceWithAppInfoFile(name, trainingType, appInfoFile, namespace string) error {
+	binary, err := exec.LookPath(kubectlCmd[0])
+	if err != nil {
+		return err
+	}
+	errs := []string{}
+
+	// get training job
+	args := []string{binary, "get", trainingType, name, "--namespace", namespace, "-o json"}
+	cmd := exec.Command("bash", "-c", strings.Join(args, " "))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		errs = append(errs, string(out))
+	}
+
+	obj := &unstructured.Unstructured{}
+	err = json.Unmarshal(out, obj)
+	if err != nil {
+		return err
+	}
+
+	patch := fmt.Sprintf(`-p='[{"op": "add", "path": "/metadata/ownerReferences", `+
+		`"value": [{"apiVersion": "%s","kind": "%s","name": "%s","uid": "%s","blockOwnerDeletion": true,"controller": true}]}]'`,
+		obj.GetAPIVersion(), obj.GetKind(), name, obj.GetUID())
+
+	data, err := ioutil.ReadFile(appInfoFile)
+	if err != nil {
+		return err
+	}
+	resources := strings.Split(string(data), "\n")
+	// add configmap
+	configmapName := fmt.Sprintf("%v-%v", name, trainingType)
+	resources = append(resources, "configmap/"+configmapName)
+
+	var wg sync.WaitGroup
+	locker := new(sync.RWMutex)
+	for _, r := range resources {
+		resource := r
```

Comment:
I suggest using `for` loop rather than ` goroutine concurrency`. And is there any way to recover if the API Server is not available in short time?

Reply:
Done